src/bot/integrations/tipeeestream.ts

```diff
@@ -0,0 +1,182 @@
+
+import { getRepository } from 'typeorm';
+import { User, UserTipInterface } from '../database/entity/user';
+import api from '../api.js';
+import events from '../events.js';
+import users from '../users.js';
+import eventlist from '../overlays/eventlist.js';
+import currency from '../currency';
+import alerts from '../registries/alerts.js';
+
+class TipeeeStream extends Integration {
+  socketToTipeeestream: SocketIOClient.Socket | null = null;
+
+  @settings()
+  @ui({ type: 'text-input', secret: false })
+  username = '';
+
+  @settings()
+  @ui({ type: 'text-input', secret: true })
+  apiKey = '';
+
+  constructor () {
+    super();
+  }
+
+  @onStartup()
+  @onChange('enabled')
+  onStateChange (key: string, val: boolean) {
+    if (val) {
+      this.connect();
+    } else {
+      this.disconnect();
+    }
+  }
+
+  async disconnect () {
+    if (this.socketToTipeeestream !== null) {
+      this.socketToTipeeestream.removeAllListeners();
+      this.socketToTipeeestream.disconnect();
+      this.socketToTipeeestream = null;
+    }
+  }
+
+  @onChange('apiKey')
+  async connect () {
+    this.disconnect();
+
+    if (this.username.trim() === '' || this.apiKey.trim() === '' || !this.enabled) {
+      return;
+    }
+
+    // get current avaliable host and port
+    // destucted response from https://api.tipeeestream.com/v2.0/site/socket
+    // example response: { "code": 200, "message": "success", "datas": { "port": "443", "host": "https://sso-cf.tipeeestream.com" } }
+    const { data: { datas: { host, port }}} = await axios.get('https://api.tipeeestream.com/v2.0/site/socket');
+
+    this.socketToTipeeestream = io(`${host}:${port}`,
+      {
+        reconnection: true,
+        reconnectionDelay: 1000,
+        reconnectionDelayMax: 5000,
+        reconnectionAttempts: Infinity,
+        query: {
+          access_token: this.apiKey.trim(),
+        },
+      });
+
+    if (this.socketToTipeeestream !== null) {
+      this.socketToTipeeestream.on('connect_error', (e) => {
+        error(chalk.red('TIPEEESTREAM.COM:') + ' error while connecting, ' + e);
+      });
+      this.socketToTipeeestream.on('connect', () => {
+        if (this.socketToTipeeestream !== null) {
+          this.socketToTipeeestream.emit('join-room', { room: this.apiKey.trim(), username: this.username.trim() });
+        }
+        info(chalk.yellow('TIPEEESTREAM.COM:') + ' Successfully connected socket to service');
+      });
+      this.socketToTipeeestream.on('reconnect_attempt', () => {
+        info(chalk.yellow('TIPEEESTREAM.COM:') + ' Trying to reconnect to service');
+      });
+      this.socketToTipeeestream.on('disconnect', () => {
+        info(chalk.yellow('TIPEEESTREAM.COM:') + ' Socket disconnected from service');
+        this.disconnect();
+        this.socketToTipeeestream = null;
+      });
+
+      this.socketToTipeeestream.on('new-event', async (data) => {
+        if (data.event.type !== 'donation') {
```

Comment:
Can we move this whole section to own function? (would be better or testing in future)